gef-ui/server/controllers/unissued-facilities/index.test.js

```diff
@@ -192,7 +195,88 @@ describe('postChangeUnissuedFacility()', () => {
     expect(api.updateFacility).toHaveBeenCalledWith('xyz', {
       coverEndDate: format(tomorrow, 'MMMM d, yyyy'),
       coverStartDate: format(tomorrow, 'MMMM d, yyyy'),
-      issueDate: format(tomorrow, 'MMMM d, yyyy'),
+      issueDate: format(now, 'MMMM d, yyyy'),
+      shouldCoverStartOnSubmission: null,
+      monthsOfCover: null,
+      name: 'Foundry4',
+      hasBeenIssued: true,
+      canResubmitIssuedFacilities: true,
+      coverDateConfirmed: true,
+    },
+    { message: 'Foundry4 is updated' },
+    '/gef/application-details/123/unissued-facilities');
+  });
+
+  it('should not update facility if issue date before submissionDate', async () => {
+    mockRequest.body.facilityType = 'CASH';
+    mockRequest.body.facilityName = 'Foundry4';
+    mockRequest.query.saveAndReturn = 'true';
+    mockRequest.body.shouldCoverStartOnSubmission = false;
+    mockRequest.body['issue-date-day'] = format(twoDaysAgoMidnight, 'd');
+    mockRequest.body['issue-date-month'] = format(twoDaysAgoMidnight, 'M');
+    mockRequest.body['issue-date-year'] = format(twoDaysAgoMidnight, 'yyyy');
+
+    mockRequest.body['cover-start-date-day'] = format(now, 'd');
+    mockRequest.body['cover-start-date-month'] = format(now, 'M');
+    mockRequest.body['cover-start-date-year'] = format(now, 'yyyy');
+
+    mockRequest.body['cover-end-date-day'] = format(tomorrow, 'd');
+    mockRequest.body['cover-end-date-month'] = format(tomorrow, 'M');
+    mockRequest.body['cover-end-date-year'] = format(tomorrow, 'yyyy');
+
+    await postChangeUnissuedFacility(mockRequest, mockResponse);
+
+    expect(mockResponse.render).toHaveBeenCalledWith('partials/unissued-change-about-facility.njk', expect.objectContaining({
+      errors: {
+        errorSummary: [
+          {
+            text: 'The issue date must not be before the date of the inclusion notice submission date',
+            href: '#issueDate',
+          }],
+        fieldErrors: {
+          issueDate: { text: 'The issue date must not be before the date of the inclusion notice submission date' },
+        },
+      },
+    }));
+
+    // should not go ahead with call as errors
+    expect(api.updateFacility).not.toHaveBeenCalledWith(MOCKS.MockRequestUnissued.facilityId, {
+      coverEndDate: format(tomorrow, 'MMMM d, yyyy'),
+      coverStartDate: format(twoDaysAgoMidnight, 'MMMM d, yyyy'),
+      issueDate: format(twoDaysAgoMidnight, 'MMMM d, yyyy'),
+      shouldCoverStartOnSubmission: null,
+      monthsOfCover: null,
+      name: 'Foundry4',
+      hasBeenIssued: true,
+      canResubmitIssuedFacilities: true,
+      coverDateConfirmed: true,
+    },
+    { message: 'Foundry4 is updated' },
```

Comment:
We should move away from using "Foundry4" in mocks 😄

Reply:
done